src/app/models/Harvest.ts

```diff
@@ -152,6 +159,20 @@ export class Harvest extends AbstractModel implements IHarvest {
       `://${this.uploadUser}:${this.uploadPassword}@`
     );
   }
+
+  public get uploadArguments(): HarvestUploadConnectionArguments {
+    return {
+      user: this.uploadUser,
+      // matches string of any character except colon after "sftp://"
+      host: this.uploadUrl.match(/sftp:\/\/([^:]+)/)[1],
+      // matches digits at the end after a colon
+      port: this.uploadUrl.match(/:([0-9]+)$/)[1],
```

Comment:
combine regexs and cache calculated values. These values could be added to the harvest model.

Reply:
done